File: GO(SECONDARY WORK)/src/Services/CustomerService/internal/types/Customer.go
package types

import (
	"time"
)

type Customer struct {
	Id        string    `bson:"_id" json:"_id" validate:"isdefault"`
	Name      string    `bson:"name" json:"name" validate:"required"`
	Address   Address   `bson:"address" json:"address" validate:"required"`
	Email     string    `bson:"email" json:"email" validate:"required,email"`
	CreatedAt time.Time `bson:"createdAt" json:"createdAt" validate:"isdefault"`
	UpdatedAt time.Time `bson:"updatedAt" json:"updatedAt" validate:"isdefault"`
}

File: C#(MAIN WORK)/src/CustomerDomain/Entities/Customer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Runtime.CompilerServices;
using CustomerDomain.ValueObjects;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace CustomerDomain.Entities
{
    //Validations are not here not to be against to SRP.
    public class Customer
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public Address Address { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}